pkg/assemblers/device-manager_test.go

```diff
@@ -0,0 +1,596 @@
+package assemblers
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/lamassuiot/lamassuiot/v2/pkg/clients"
+	"github.com/lamassuiot/lamassuiot/v2/pkg/config"
+	"github.com/lamassuiot/lamassuiot/v2/pkg/models"
+	"github.com/lamassuiot/lamassuiot/v2/pkg/resources"
+	"github.com/lamassuiot/lamassuiot/v2/pkg/services"
+	"github.com/lamassuiot/lamassuiot/v2/pkg/storage/postgres"
+	postgres_test "github.com/lamassuiot/lamassuiot/v2/pkg/test/subsystems/storage/postgres"
+)
+
+type DeviceManagerTestServer struct {
+	Service              services.DeviceManagerService
+	HttpDeviceManagerSDK services.DeviceManagerService
+	BeforeEach           func() error
+	AfterSuite           func()
+}
+
+func initPostgres(dbs []string) (config.PostgresPSEConfig, postgres_test.PostgresSuite) {
+	return postgres_test.BeforeSuite(dbs)
+}
+
+func buildCASimpleTestServer(pConfig config.PostgresPSEConfig, postgresSuite postgres_test.PostgresSuite) (*CATestServer, error) {
```

Comment:
Could we reuse the one in ca_test, since the CA service is a dependency of the device manager?

Reply:
The ca_test's BuildCASimpleTestServer configures 2 cryptoengines and starts Vault container which is not required for these tests.

Reply:
Maybe we could add a boolean variable to decide if more than one engines are needed? I would rather have only one "buildCATestServer"

Reply:
I've moved the assembling logic to a common place, now it its shared by all the services tests. 

The assembling process aggregates the BeforeEach and AfterSuite logic from each service. It is exposed as a single beforeEach and afterSuite functions.